### PR TITLE
Cross browser fixes

### PR DESCRIPTION
**Pull request by iondrimba**
**CHANGELOG** :memo:

* Added various crossbrowser fixes
* Changed autoprefixer from CSSNano to Postcss
* Replaced flexbox mixins with native CSS

### CODE REVIEW

File: postcss.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    "browsers": [
+      "last 2 versions",
+      "safari >= 7",
+      "ie >= 9",
+      "ff >= 20",
+      "ios >=7",
+      "android >=4"
+    ]
+  }
+}
```
Comment (eduardojmatos) [inline postcss.json]: @iondrimba what do you think about don't uses `postcss.json` and use a `browserslist` config? Example:

```
# browserslist
# Browsers that we support

Last 2 versions
Safari >= 7
ie >= 9
FF >= 20
iOS >=7
Android >=4
```

Whitout any extension, like documentation suggest https://github.com/ai/browserslist#browserslist.

This is more clean, what do you think?

**Comment by iondrimba** [comment]
[(5) Ajustes no site - IE 11/10, iOS 8/7, Android 4.3+, Safari](https://trello.com/c/DueSzOte/264-5-ajustes-no-site-ie-11-10-ios-8-7-android-4-3-safari)

File: src/scss/components/_dialog.scss [hunk 2]
```diff
@@ -29,13 +29,14 @@
       border-right: 15px solid get-color(breeze);
       border-bottom: 15px solid get-color(breeze);
       border-left: 15px solid get-color(breeze);
+      -ms-transform: translateX(-50%);
```
Comment (iondrimba) [inline src/scss/components/_dialog.scss]: No it don't includes -ms for transform

File: src/scss/components/_list.scss [hunk 3]
```diff
@@ -40,6 +40,7 @@
       &:hover,
       &:active,
       &:focus {
+        -ms-transform: translateX(8px);
```
Comment (iondrimba) [inline src/scss/components/_list.scss]: Not covered by autoprefixer

File: src/scss/components/_picture.scss [hunk 4]
```diff
@@ -52,8 +52,9 @@
       position: relative;
       left: 50%;
       min-height: 100%;
-      max-width: initial;
+      max-width: none;
```
Comment (iondrimba) [inline src/scss/components/_picture.scss]: I testes in all latest browsers and the legacy ones. Did not impact.

File: src/scss/components/_picture.scss [hunk 5]
```diff
@@ -52,8 +52,9 @@
...
       min-width: 100%;
+      -ms-transform: translateX(-50%);
```
Comment (iondrimba) [inline src/scss/components/_picture.scss]: Not covered by autoprefixer

File: src/scss/components/_price-variation.scss [hunk 6]
```diff
@@ -7,13 +7,16 @@
 **/
 
 .price-variation {
-  @include flexbox;
-  @include flex-direction(row);
-  @include flex-wrap(wrap);
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  position: relative;
```
Comment (iondrimba) [inline src/scss/components/_price-variation.scss]: Yes, Safari breaks

File: src/scss/components/_price-variation.scss [hunk 7]
```diff
@@ -7,13 +7,16 @@
...
 
   &__item {
-    @include flex;
-    @include flex-direction(row);
+    display: flex;
+    flex-direction: column;
+    flex: 1;
+    width: 50%;
```
Comment (iondrimba) [inline src/scss/components/_price-variation.scss]: Yes, Safari breaks

File: src/scss/components/_thumbnail.scss [hunk 8]
```diff
@@ -29,7 +29,9 @@
     flex-direction: column-reverse;
 
     &:hover {
+      transition: -ms-transform .2s linear;
       transition: transform .2s linear;
+      -ms-transform: translateY(-7px) scale(1.03);
```
Comment (iondrimba) [inline src/scss/components/_thumbnail.scss]: Not covered by autoprefixer

File: src/scss/components/_widget.scss [hunk 9]
```diff
@@ -11,15 +11,16 @@
   border-radius: 7px;
   border: 1px solid get-color((color: blackNeutral, opacity: .15));
   background-color: get-color(air);
+  position: relative;
```
Comment (iondrimba) [inline src/scss/components/_widget.scss]: Crossbrowser

File: src/scss/components/_widget.scss [hunk 10]
```diff
@@ -11,15 +11,16 @@
...
   &--over-banner {
     margin-top: -180px;
   }
 
   &__title {
-    margin: 14px 10%;
+    margin: 14px auto;
     padding: 15px 0;
-    width: 100%;
+    width: 80%;
```
Comment (iondrimba) [inline src/scss/components/_widget.scss]: Yes

File: src/scss/generic/_reset.scss [hunk 11]
```diff
@@ -12,3 +12,10 @@ em {
 strong {
   font-weight: 700;
 }
+
+select,
+input,
+button,
+textarea {
+  font-family: inherit;
+}
```
Comment (iondrimba) [inline src/scss/generic/_reset.scss]: I think it could be, its to minor to create a specific Gaiden release

File: postcss.json (hunk 1 above)
Comment (iondrimba) [inline postcss.json]: I think having Postcss is a good tool to have, we could use it for other improvements in CSS

**Comment by iondrimba** [comment]
@eduardojmatos All changes in this PR is to fix inconsistency across browsers. 
The Price Card fix is a problem only in Safari so it is crossbrowser too.

What fixes you dont think should be in this PR?

File: postcss.json [hunk 12]
```diff
@@ -0,0 +1,12 @@
+{
+  "autoprefixer": {
+    "browsers": [
+      "last 2 versions",
+      "safari >= 7",
+      "ie >= 9",
+      "ff >= 20",
+      "ios >=7",
+      "android >=4"
```
Comment (iondrimba) [inline postcss.json]: It will concat all prefixes

File: src/scss/components/_media_box.scss [hunk 13]
```diff
@@ -27,24 +27,22 @@
...
-
+    flex-shrink: 0;
     padding-right: 15px;
   }
 
   &__content {
-    @include flex-wrap(wrap);
-    @include flex-direction(column);
+    flex-wrap: wrap;
+    flex-direction: row;
```
Comment (iondrimba) [inline src/scss/components/_media_box.scss]: Very strange but it was.

File: dist/gaiden.min.css [hunk 14]
```diff
@@ -1,2 +1,14 @@
...
...ol-push-small-4{margin-left:34.58333%}.col-offset-small-4{left:33.33333%}.col-small-5{width:39.16667%}.col-push-small-5{margin-left:42.91667%}.col-offset-small-5{left:41.66667%}.col-small-6{width:47.5%}.col-push-small-6{margin-left:51.25%}.col-offset-small-6{left:50%}.col-small-7{width:55.83333%}.col-push-small-7{margin-left:59.58333%}.col-offset-small-7{left:58.33333%}.col-small-8{width:64.16667%}.col-push-small-8{margin-left:67.91667%}.col-offset-small-8{left:66.66667%}.col-small-9{width:72.5%}.col-push-small-9{margin-left:76.25%}.col-offset-small-9{left:75%}.col-small-10{width:80.83333%}.col-push-small-10{margin-left:84.58333%}.col-offset-small-10{left:83.33333%}.col-small-11{width:89.16667%}.col-push-small-11{margin-left:92.91667%}.col-offset-small-11{left:91.66667%}.col-small-12{width:97.5%}.col-push-small-12{margin-left:101.25%}.col-offset-small-12{left:100%}.space-section-small{margin-top:60px}.space-section-medium{margin-top:72px}.space-section-large{margin-top:80px}.space-box-small{margin-top:24px}.space-box-medium{margin-top:36px}.space-box-large{margin-top:48px}.space-element-small{margin-top:6px}.space-element-medium{margin-top:12px}.space-element-large{margin-top:16px}
```
Comment (iondrimba) [inline dist/gaiden.min.css]: @eduardojmatos  Should we add them to .gitignore?

File: postcss.json (hunk 1 above)
Comment (iondrimba) [inline postcss.json]: I will try it out!

**Comment by iondrimba** [comment]
@eduardojmatos Removing postcss.json worked :+1:

File: src/scss/components/_card.scss [hunk 15]
```diff
@@ -24,12 +24,12 @@
...
 .card {
-  @include flex($fg: 1);
-
-  padding: 24px;
   background-color: get-color(air);
   border-radius: $borderRadius;
   box-shadow: 0 4px 8px 0 get-color((color: blackNeutral, opacity: .12));
+  flex: 1 1 auto;
+  padding: 24px;
+  width: inherit;
```
Comment (iondrimba) [inline src/scss/components/_card.scss]: The padding just moved from location. Width was for crossbrowser fix

File: src/scss/components/_dialog.scss (hunk 2 above)
Comment (eduardojmatos) [inline src/scss/components/_dialog.scss]: 😞

File: src/scss/components/_media_box.scss (hunk 13 above)
Comment (eduardojmatos) [inline src/scss/components/_media_box.scss]: ok 😭

File: src/scss/components/_picture.scss (hunk 4 above)
Comment (eduardojmatos) [inline src/scss/components/_picture.scss]: Fair 👍

File: src/scss/components/_price-variation.scss (hunk 6 above)
Comment (eduardojmatos) [inline src/scss/components/_price-variation.scss]: Ok. Let's check on another browsers if there is no side effect (triborg have this component)

File: src/scss/components/_price-variation.scss (hunk 7 above)
Comment (eduardojmatos) [inline src/scss/components/_price-variation.scss]: [Some here](https://github.com/getninjas/gaiden/pull/86#discussion_r107519418)

File: src/scss/components/_widget.scss (hunk 10 above)
Comment (eduardojmatos) [inline src/scss/components/_widget.scss]: @iondrimba maybe we must test on another browsers to check side effects

File: dist/gaiden.min.css (hunk 14 above)
Comment (eduardojmatos) [inline dist/gaiden.min.css]: @iondrimba I think that not. Dist files is used as a official version of Gaiden and will be used on our document page (on `gh-pages` branch).

File: src/scss/components/_price-variation.scss (hunk 6 above)
Comment (iondrimba) [inline src/scss/components/_price-variation.scss]: @eduardojmatos I checked on Triborg, this fix is for Tribrog

File: src/scss/components/_widget.scss (hunk 10 above)
Comment (iondrimba) [inline src/scss/components/_widget.scss]: @eduardojmatos I the following checks Chrome, FF, Safari, EDGE, IE 11/10, iOS 9/8/7 Android 4.3-

**Comment by eduardojmatos** [comment]
@iondrimba my suggestion is open a PR for:

- `src/scss/generic/_reset.scss` changes;
- all fixes that have side effects on another browsers (like `width: 80%`)

**Comment by eduardojmatos** [comment]
@iondrimba restating, if there is not side effect on UI, let's move on!

**Comment by iondrimba** [comment]
@eduardojmatos removed font-family config for another PR

File: src/scss/components/_widget.scss (hunk 10 above)
Comment (eduardojmatos) [inline src/scss/components/_widget.scss]: Nice! Let's continue the game 👾

File: src/scss/generic/_reset.scss (hunk 11 above)
Comment (iondrimba) [inline src/scss/generic/_reset.scss]: removed like suggested

File: src/scss/components/_widget.scss (hunk 10 above)
Comment (iondrimba) [inline src/scss/components/_widget.scss]: @eduardojmatos I'm going to revert this change, because in smaller sizes the title doesn't fit the 100% space. (mobile)

File: src/scss/components/_thumbnail.scss [hunk 16]
```diff
@@ -25,14 +25,12 @@
   will-change: transform;
 
   &--action {
-    display: flex;
-    flex-direction: column-reverse;
+    display: block;
+    transform: scale(-1);
```
Comment (eduardojmatos) [inline src/scss/components/_thumbnail.scss]: @iondrimba `scale(-1)` here will not affect another browsers?

Comment (iondrimba) [inline src/scss/components/_thumbnail.scss]: @eduardojmatos Tested in all major browsers 👍
